pythclient/hermes.py

```diff
@@ -0,0 +1,173 @@
+import asyncio
+from typing import TypedDict
+
+import httpx
+import os
+
+from .price_feeds import Price
+
+HERMES_ENDPOINT_HTTPS = "https://hermes.pyth.network/api/"
+HERMES_ENDPOINT_WSS = "wss://hermes.pyth.network/ws"
+
+
+class PriceFeed(TypedDict):
+    feed_id: str
+    price: Price
+    ema_price: Price
+    vaa: str
+
+
+
+class HermesClient:
+    def __init__(self, feed_ids: list[str], endpoint=HERMES_ENDPOINT_HTTPS, ws_endpoint=HERMES_ENDPOINT_WSS):
+        self.feed_ids = feed_ids
+        self.pending_feed_ids = feed_ids
+        self.prices_dict: dict[str, PriceFeed] = {}
+        self.client = httpx.AsyncClient()
```

Comment:
the `httpx.AsyncClient` is instantiated but not explicitly closed, which could potentially lead to resource leaks, maybe use the client as a context manager to ensure it's properly closed e.g.
```
    async with self.client as client:
        data = (await client.get(url)).json()
```

Reply:
The issue is that with context manager usage, any one call will close the client which makes it impossible to open again. I think using `asyncio.run` manages the resource leakage by closing all tasks running within the event loop whenever it's shut down: https://github.com/python/cpython/blob/c0b0c2f2015fb27db4306109b2b3781eb2057c2b/Lib/asyncio/runners.py#L64-L79